app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -3043,48 +3071,29 @@ private void onMetadataChanged(@NonNull final MediaSourceTag tag) {
         }
     }
 
-    private void maybeUpdateCurrentMetadata() {
+    private void updateMetadataWith(@NonNull final StreamInfo streamInfo) {
         if (exoPlayerIsNull()) {
             return;
         }
 
-        final MediaSourceTag metadata;
-        try {
-            final MediaItem currentMediaItem = simpleExoPlayer.getCurrentMediaItem();
-            if (currentMediaItem == null || currentMediaItem.playbackProperties == null
-                    || currentMediaItem.playbackProperties.tag == null) {
-                return;
-            }
-            metadata = (MediaSourceTag) currentMediaItem.playbackProperties.tag;
-        } catch (final IndexOutOfBoundsException | ClassCastException ex) {
-            if (DEBUG) {
-                Log.d(TAG, "Could not update metadata", ex);
-            }
-            return;
-        }
-
-        maybeAutoQueueNextStream(metadata);
-
-        if (currentMetadata == metadata) {
-            return;
-        }
-        currentMetadata = metadata;
-        onMetadataChanged(metadata);
+        maybeAutoQueueNextStream(streamInfo);
+        onMetadataChanged(streamInfo);
+        NotificationUtil.getInstance().createNotificationIfNeededAndUpdate(this, true);
```

Comment:
Also, why does this now have `forceRecreate = true`? Also, it seems like you are calling `createNotificationIfNeededAndUpdate` twice when `updateMetadataWith` is called, since there is a `createNotificationIfNeededAndUpdate(this, false);` also inside `onMetadataChanged`. And as you can see that one has `forceRecreate = false`.

Reply:
Oversight on my part. `createNotificationIfNeededAndUpdate ` in `onMetadataChanged` should be removed.